docs/dev/hacking-guide.md

```diff
@@ -0,0 +1,193 @@
+
+```bash
+IMG="quay.io/your-repo/cluster-cloud-controller-manager-operator:<branch-name>" make images
+docker push quay.io/your-repo/cluster-cloud-controller-manager-operator:<branch-name>
+```
+
+Second step, similar to running the component locally we need to scale down CVO.
+
+```bash
+oc scale --replicas=0 deployment/cluster-version-operator -n openshift-cluster-version
+```
+
+Next, edit `cloud-controller-manager-images` ConfigMap in namespace `openshift-cloud-controller-manager-operator` and place a link to your image there.
+
+The `openshift-cloud-controller-manager-operator` project contains all the resources and components that are used by the CCCMO.
+There is a ConfigMap named `cloud-controller-manager-images`.
+This ConfigMap contains references to all the images used by the CCCMO.
+It uses these images to deploy the CCM to various platforms and ensure that the running images are from the correct release payload.
+
+The ConfigMap looks something like this:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cloud-controller-manager-images
+  namespace: openshift-cloud-controller-manager-operator
+data:
+  images.json: |
+    {
+      "cloudControllerManagerAWS": "registry.ci.openshift.org/openshift:aws-cloud-controller-manager",
+      "cloudControllerManagerOpenStack": "registry.ci.openshift.org/openshift:openstack-cloud-controller-manager"
+    }
+```
+
+You could edit existing ConfigMap or create a copy with your images and apply in the cluster.
+
+```bash
+oc edit configmap/cloud-controller-manager-images -n openshift-cloud-controller-manager-operator
+```
+
+With the new image information loaded into the ConfigMap, the next thing you might do is replace the CCCMO. This operator controls how the specific cloud controllers are deployed and coordinated. You only change this component if there is something you are testing.
+
+The easiest way to change this operator is to change the image reference in the deployment. You can use the following command to edit image references or other things in the deployment:
+
+```bash
+oc edit deployment/cluster-cloud-controller-manager -n openshift-cloud-controller-manager-operator
+```
+
+If you want images ConfigMap changes to reconcile you need to cause creation of new replicaSet for CCM, which will ensure the new content of ConfigMap is mounted.
+
+After changing the download reference in the images ConfigMap an easy way to swap out the controller is to let the CCCMO do it for you. You can delete the deployment associated with the cloud provider CCM and then the CCCMO will create a new one for you, like this:
+
+```bash
+oc delete deployment/aws-cloud-controller-manager -n openshift-cloud-controller-manager
+```
+
+## How to build operator in combination with other operators and build a release image
+
+At the current stage of development it is frequent that for testing a feature, another openshift operator change should be in place to make the feature work in CCCMO.
```

Comment:
this could probably use a little more explanation, i'm not sure which operator change you are talking about when you mention "another openshift operator change should be in place". 

or is this just saying that _in general_ you might need to make other changes?

Reply:
In general. This operator lives between multiple other components, and so the changes in one place create changes in another.